master/buildbot/www/avatar.py

```diff
@@ -34,12 +37,143 @@ def getUserAvatar(self, email, username, size, defaultAvatarUrl):
 class AvatarGitHub(AvatarBase):
     name = "github"
 
-    AVATAR_URL = 'https://avatars.githubusercontent.com/'
+    DEFAULT_GITHUB_API_URL = 'https://api.github.com'
 
-    def getUserAvatar(self, email, username, size, defaultAvatarUrl):
-        github_url = self.AVATAR_URL + username.decode("utf-8")
-        raise resource.Redirect(github_url)
+    def __init__(self,
+                 github_api_endpoint=None,
+                 token=None,
+                 debug=False,
+                 verify=False):
+        httpclientservice.HTTPClientService.checkAvailable(self.__class__.__name__)
+
+        self.github_api_endpoint = github_api_endpoint
+        if github_api_endpoint is None:
+            self.github_api_endpoint = self.DEFAULT_GITHUB_API_URL
+        self.token = token
+        self.debug = debug
+        self.verify = verify
+
+        self.master = None
+        self.client = None
+
+    @defer.inlineCallbacks
+    def _getHTTPClient(self):
```

Comment:
Let's prefer snake_case for any new code that is not a public API.

Reply:
Done